File: Task/M-bius-function/Julia/m-bius-function.jl
using Primes

# modified from reinermartin's PR at https://github.com/JuliaMath/Primes.jl/pull/70/files
function moebius(n::Integer)
    @assert n > 0
    m(p, e) = p == 0 ? 0 : e == 1 ? -1 : 0
    reduce(*, m(p, e) for (p, e) in factor(n) if p ≥ 0; init=1)
end
μ(n) = moebius(n)

print("First 199 terms of the Möbius sequence:\n   ")
for n in 1:199
    print(lpad(μ(n), 3), n % 20 == 19 ? "\n" : "")
end

File: Task/M-bius-function/R/m-bius-function.r
# Moebius function

Moebius <- function(n) {
  if (n == 1)
    return(1)
  m <- 1
  f <- 2
  repeat {
    if (n %% (f * f) == 0)
      m <- 0
    else {
      if (n %% f == 0) {
        m <- -m
        n <- n %/% f
      }
      f <- f + 1
    }
    if (f > n || m == 0)
      break
  }
  return(m)
}

mb <- matrix(0, 10, 10)
for (t in 0:9)
  for (u in 1:10)
    mb[t + 1, u] <- Moebius(10 * t + u)
mb
